src/openrct2-ui/WindowManager.cpp

```diff
@@ -926,6 +930,7 @@ class WindowManager final : public IWindowManager
         ColourSchemeUpdate(w);
         w->Invalidate();
         w->OnOpen();
+        w->OnResize();
```

Comment:
This should be rethought. For historical reasons, the initial widgets are often set _after_ the window is opened. That is, often not by the constructor nor the OnOpen event. It would be much better if `SetWidgets` handled this, but that's part of a different namespace right now. Perhaps we should just take `ResizeFrame` to `Window_internal.cpp`, or the other way around? (Moving `SetWidgets` away from `WindowBase` doesn't feel right, though.)

Reply:
I can’t really think of a good answer to this. I would suggest you do what seems most sensible to you.